### DISCUSSION TITLE
Convert Enum to string in OC and YesSql

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#13154] [General]
### Is your feature request related to a problem? Please describe.
Currently, we store the enum as in. Problem with this approach if someone change the enum order or adds a new value in a place other than the default, it'll cause problem. Also, if someone is looking at the raw json file like in the workflow for example, it's hard to understand that the int value means.

### Describe the solution you'd like
Convert Enum to string in OC and YesSql

**Comment by sebastienros** [comment]
Other option, always set associated values to enums. And don't change them or reuse them. Because otherwise you have the other issue that you'd want to rename it and would break the data.

**Reply by MikeAlhayek** [reply]
Yes that would solve one of the problems. But, storing value instead of string will still make it not readable. Just if you are reading the JSON string you won't not know what the integer means.

**Comment by hyzx86** [comment]
Explicitly specify values of enumeration types before using custom serialization properties

``` C# 
Read=1,
Write=2,
ReadAndWrite=3
```
To serialize a particular property into an object, for example:
``` json
{ FilePermission: {"1":"Read"} }
```
When deserializing, only the Key value is taken

For compatibility with older versions, we can also handle compatibility in a custom serialization feature
For example, check whether the Value of a property is JObject
